flink/src/main/java/org/apache/iceberg/flink/data/FlinkOrcWriters.java

```diff
@@ -245,8 +245,19 @@ public void nonNullWrite(int rowId, ArrayData data, ColumnVector output) {
       ListColumnVector cv = (ListColumnVector) output;
       cv.lengths[rowId] = data.size();
       cv.offsets[rowId] = cv.childCount;
-      cv.childCount += cv.lengths[rowId];
+      // cv.childCount is for some reason an int, which generates all of these
+      // NarrowingCompoundAssignment warnings. Although this does nothing to prevent
+      // overflow from adding too many values to cv.childCount because `ListColumnVector`
+      // comes from package `org.apache.orc.storage.ql.exec.vector`, it at least removes
```

Comment:
As it is, my point is we can't necessarily change it to be a long unless somebody on the orc side can do that (which I think is a possible bug given that the arrays `lengths` and `offsets` are both long arrays).

Reply:
I think the solution to use the int source data makes sense. We probably don't need to have the long comment, since it is clear that this is correct.